shared/contracts/BaseTemplate.sol

```diff
@@ -56,155 +62,178 @@ contract BaseTemplate is APMNamehash, IsContract {
+        return _installApp(_dao, _appId, new bytes(0), false);
     }
 
-    function _installDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
-        return _installApp(dao, appId, new bytes(0), true);
+    function _installDefaultApp(Kernel _dao, bytes32 _appId) internal returns (address) {
+        return _installApp(_dao, _appId, new bytes(0), true);
     }
 
-    function _installApp(Kernel dao, bytes32 appId, bytes data, bool setDefault) internal returns (address) {
-        address latestBaseAppAddress = _latestVersionAppBase(appId);
-        address instance = address(dao.newAppInstance(appId, latestBaseAppAddress, data, setDefault));
-        emit InstalledApp(instance, appId);
+    function _installApp(Kernel _dao, bytes32 _appId, bytes _data, bool _setDefault) internal returns (address) {
+        address latestBaseAppAddress = _latestVersionAppBase(_appId);
+        address instance = address(_dao.newAppInstance(_appId, latestBaseAppAddress, _data, _setDefault));
+        emit InstalledApp(instance, _appId);
         return instance;
     }
 
-    function _latestVersionAppBase(bytes32 appId) internal view returns (address base) {
-        Repo repo = Repo(PublicResolver(ens.resolver(appId)).addr(appId));
+    function _latestVersionAppBase(bytes32 _appId) internal view returns (address base) {
+        Repo repo = Repo(PublicResolver(ens.resolver(_appId)).addr(_appId));
         (,base,) = repo.getLatest();
     }
 
     /* TOKEN */
 
-    function _createToken(string name, string symbol) internal returns (MiniMeToken) {
-        require(isContract(address(miniMeFactory)), "TEMPLATE_MINIME_FACTORY_NOT_PROVIDED");
-        MiniMeToken token = miniMeFactory.createCloneToken(MiniMeToken(address(0)), 0, name, 18, symbol, true);
+    function _createToken(string _name, string _symbol) internal returns (MiniMeToken) {
+        require(address(miniMeFactory) != address(0), ERROR_MINIME_FACTORY_NOT_PROVIDED);
+        MiniMeToken token = miniMeFactory.createCloneToken(MiniMeToken(address(0)), 0, _name, 18, _symbol, true);
         emit DeployToken(address(token));
         return token;
     }
 
+    function _ensureMiniMeFactoryIsValid(address _miniMeFactory) internal view {
+        require(isContract(address(_miniMeFactory)), ERROR_MINIME_FACTORY_NOT_CONTRACT);
+    }
+
     /* IDS */
 
-    function _registerID(string name, address owner) internal {
-        require(isContract(address(aragonID)), "TEMPLATE_ARAGON_ID_NOT_PROVIDED");
-        aragonID.register(keccak256(abi.encodePacked(name)), owner);
+    function _registerID(string _name, address _owner) internal {
+        require(address(aragonID) != address(0), ERROR_ARAGON_ID_NOT_PROVIDED);
+        aragonID.register(keccak256(abi.encodePacked(_name)), _owner);
+    }
+
+    function _ensureAragonIdIsValid(address _aragonID) internal view {
+        require(isContract(address(_aragonID)), ERROR_ARAGON_ID_NOT_CONTRACT);
```

Comment:
Here and in `_ensureMiniMeFactoryIsValid()`, we should allow `address(0)` as well (`isContract` will return false if `address(0)` is given).

Reply:
These helpers are used actually to check that those dependencies are valid: non-zero and a contract. All the templates using them have to check that, otherwise, their creation methods will revert.

Reply:
Ah got it, that makes sense.